front-end/src/main/kotlin/org/jetbrains/kotlin/spec/entity/test/parameters/testArea/TestArea.kt

```diff
@@ -0,0 +1,61 @@
+package org.jetbrains.kotlin.spec.entity.test.parameters.testArea
+
+import org.jetbrains.kotlin.spec.entity.test.TestCase
+import org.jetbrains.kotlin.spec.entity.test.parameters.testArea.SpecTestsParser.implementationTestInfoPattern
+import org.jetbrains.kotlin.spec.entity.test.parameters.testArea.SpecTestsParser.topLevelDirectivePattern
+
+enum class TestArea(val path: String,
+                    val description: String,
+                    val attributeValue: String,
+                    val testCasesParser: (String) -> List<TestCase>
+) {
+    Diagnostics("diagnostics", "Front-end diagnostics tests", "diagnostics",
+            { data -> parseDiagnosticsTest(data) }),
+    CodegenBox("codegen/box", "Codegen box tests", "box",
+            { data -> parseCodegenBoxTest(data) }
+    );
+
+    companion object {
+        fun parseCodegenBoxTest(testFileCode: String): List<TestCase> {
+            val data = removeMetadataComments(testFileCode)
```

Comment:
I think it'd be better to unify terminology: choose `metadataComments` or `testInfoElements`

Reply:
testInfoElements are located in json, metadataComments are located in .kt files